src/MetaMorpho.sol

```diff
@@ -126,6 +131,7 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
         if (morpho == address(0)) revert ErrorsLib.ZeroAddress();
 
         MORPHO = IMorpho(morpho);
+        DECIMALS_OFFSET = uint8(uint256(18).zeroFloorSub(IERC20Metadata(_asset).decimals()));
```

Comment:
Are we sure there is a use case for tokens with more than 18 decimals ? Otherwise we could do something like:
```suggestion
        DECIMALS_OFFSET = uint8(18 - IERC20Metadata(_asset).decimals());
```

Reply:
there's no large usecase (PEPE?) I have in mind but we could have ERC4626 with more than 18 decimals as loan asset (as almost we did) so I'd rather keep it

Reply:
exact